### PR TITLE
feat(ingest): allow for deleting aspects in cli with -p tag

### PR DESCRIPTION
**Pull request by Anirudh181001**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by Anirudh181001** [comment]
In the datahub docs, the following example is included:
```
# Delete all of the aspect values for a given platform and a timeseries aspect.
datahub delete -p "<platform>" -a "<timeseries aspect>" --hard
Eg: datahub delete -p "snowflake" -a "datasetProfile" --hard
```
However, there is a small bug in which the aspect information is not passed on when a delete command is run with the -p flag. Moreover, this command also cannot be run without the --hard flag since there is no soft delete for an aspect. 
This PR addresses the above concerns and provide a simple fix for it.

File: metadata-ingestion/src/datahub/cli/delete_cli.py [hunk 1]
```diff
@@ -388,6 +391,10 @@ def _delete_one_urn(
     deletion_result.num_records = UNKNOWN_NUM_RECORDS  # Default is unknown
 
     if soft:
+        if aspect_name: 
+            raise click.UsageError(
+            "Please providem --hard flag, as aspect values cannot be soft deleted."
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/cli/delete_cli.py:396]:
```suggestion
            "Please provide --hard flag, as aspect values cannot be soft deleted."
```

**Comment by hsheth2** [comment]
@Anirudh181001 - did you test this locally?

**Comment by Anirudh181001** [comment]
> @Anirudh181001 - thanks for the PR!
> 
> It overall looks good. Given that we don't have unit tests for delete, I just wanted to check if you tested this locally?

Hi @hsheth2 I tested this locally! Seems to work fine!

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@Anirudh181001 awesome!

Looks like there's a tiny lint error - could you just run `./gradlew :metadata-ingestion:lintFix`?

**Comment by Anirudh181001** [comment]
> @Anirudh181001 awesome!
> 
> Looks like there's a tiny lint error - could you just run `./gradlew :metadata-ingestion:lintFix`?

Yup fixed now! Can I go ahead and merge?

**Pull request merged**